src/Interpreters/InterpreterInsertQuery.cpp

```diff
@@ -121,8 +121,7 @@ StoragePtr InterpreterInsertQuery::getTable(ASTInsertQuery & query)
 
             if (current_context->getSettingsRef()[Setting::allow_experimental_analyzer])
             {
-                InterpreterSelectQueryAnalyzer interpreter_select(query.select, current_context, select_query_options);
-                header_block = interpreter_select.getSampleBlock();
+                header_block = InterpreterSelectQueryAnalyzer::getSampleBlock(query.select, current_context, select_query_options);
```

Comment:
The difference here is that the static `getSampleBlock` sets `only_analyze = true` in SelectQueryOptions, which prevents query analysis from calling `IStorage::read` on the tables (which breaks `input(...)` table function in particular).